src/neo4j_genai/types.py

```diff
@@ -93,3 +99,59 @@ class SearchType(str, Enum):
 
     VECTOR = "vector"
     HYBRID = "hybrid"
+
+
+class EmbedderModel(BaseModel):
+    embedder: Optional[Any]
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @field_validator("embedder")
+    def check_embedder(cls, value):
+        if not hasattr(value, "embed_query") or not callable(
+            getattr(value, "embed_query", None)
+        ):
+            raise ValueError(
+                "Provided embedder object must have an 'embed_query' callable method."
+            )
+        return value
+
+
+class Neo4jDriverModel(BaseModel):
+    driver: neo4j.Driver
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @field_validator("driver")
+    def check_driver(cls, value):
+        if not isinstance(value, neo4j.Driver):
```

Comment:
Isn't this validation covered by the type declaration above?

Reply:
Replied in the question regarding `model_config`

Reply:
I understand it in the case of embedder, where we can have different types (maybe not ours). But here we will always have a neo4j driver right? Or did I miss something?

Reply:
Like `Embedder`, `__get_pydantic_core_schema__` is not implemented on `neo4j.Driver` as well which would result in the same error. So either we are more lenient on this type here or we try and implement the same on the python driver library. I think it is easier for now to do this here

Reply:
Oh ok, interesting! Maybe worth opening an issue for the driver to get it implemented, wdyt? 
But anyway in the meantime we can do it this way, LGTM.